frontend/src/features/admin-form/responses/FeedbackPage/FeedbackDownloadButton.tsx

```diff
@@ -1,39 +1,36 @@
-import { useMemo } from 'react'
-import { CSVLink } from 'react-csv'
+import { useCallback } from 'react'
 import { BiDownload } from 'react-icons/bi'
 
-import { ProcessedFeedbackMeta } from '~shared/types'
-
 import Button from '~components/Button'
 
+import { useFormFeedbackMutations } from '../../common/mutations'
+
 type FeedbackDownloadButtonProps = {
   isDisabled: boolean
   formId: string | undefined
-  feedback: ProcessedFeedbackMeta[] | undefined
+  formTitle: string | undefined
 }
 
 export const FeedbackDownloadButton = ({
-  feedback,
   isDisabled,
   formId,
+  formTitle,
 }: FeedbackDownloadButtonProps) => {
-  const data = useMemo(() => {
-    if (!feedback) return ''
-    return feedback?.map((entry) => ({
-      index: entry.index,
-      date: entry.date,
-      feedback: entry.comment,
-      rating: entry.rating,
-    }))
-  }, [feedback])
+  const { downloadFormFeedbackMutation } = useFormFeedbackMutations(
+    formId ?? '',
+    formTitle ?? '',
+  )
+
+  const handleClick = useCallback(
+    () => downloadFormFeedbackMutation.mutate(),
+    [downloadFormFeedbackMutation],
+  )
```

Comment:
```suggestion
  const handleClick = useCallback(
    () => {
      if (!formId || !formTitle) return
      return downloadFormFeedbackMutation.mutate(formId, formTitle)
    },
    [downloadFormFeedbackMutation],
  )
```

this will prevent the current bug where it may be possible (also not la, since its disabled) to trigger API call with empty args